spec/01-unit/01-db/08-cache_warmup_spec.lua

```diff
@@ -213,18 +214,21 @@ describe("cache_warmup", function()
 
     assert.truthy(cache_warmup.execute({"my_entity", "services"}))
 
-    ngx.sleep(0) -- yield so that async DNS caching happens
-
-    -- `my_entity` isn't a core entity; lookup is on client cache
-    assert.same(kong.cache:get("111").bbb, 222)
-    assert.same(kong.cache:get("333").bbb, 444)
-
-    assert.same(kong.core_cache:get("a").host, "example.com")
-    assert.same(kong.core_cache:get("b").host, "1.2.3.4")
-    assert.same(kong.core_cache:get("c").host, "example.test")
-
-    -- skipped IP entry
-    assert.same({ "example.com", "example.test" }, dns_queries)
+    -- waiting async DNS cacheing
+    helpers.wait_until(function ()
```

Comment:
replace wait_until with deterministic job execution, either 1) a wait_until on the executed_job counter, 2) or a "flush" function in the timerng library just to use in tests to force execute existing jobs